projects/lib/src/xboardengine.cpp

```diff
@@ -624,10 +627,14 @@ void XboardEngine::parseLine(const QString& line)
 			return;
 		}
 
-		Chess::Move move = board()->moveFromString(args);
+		// remove "..." of old format if necessary
+		int mark = (args.indexOf("..."));
+		const QString movestr = mark < 0 ? args : args.right(mark + 4);
```

Comment:
You're lucky that the move string is exactly 4 characters long; otherwise this wouldn't work. `QString::right()` returns the `n` rightmost characters so `mark + 4` is a bit misleading because it's used as an index: http://doc.qt.io/qt-5/qstring.html#right

We already checked if `args.startsWith("...")` earlier so maybe could just use that info here to set `movestr = args.mid(4)`.